### PR TITLE
Make git push options accept short name

### PR DESCRIPTION
**Pull request by wxiaoguang**
Just like what most CLI parsers do: `--opt` means `opt=true`

Then users could use `-o force-push` as `-o force-push=true`

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: routers/private/hook_post_receive.go [hunk 1]
```diff
@@ -208,7 +208,7 @@ func HookPostReceive(ctx *gitea_context.PrivateContext) {
 			return
 		}
 
-		cols := make([]string, 0, len(opts.GitPushOptions))
+		cols := make([]string, 0, 2)
```
Comment (a1012112796) [inline routers/private/hook_post_receive.go:211]: why not using len of push options?

Comment (wxiaoguang) [reply routers/private/hook_post_receive.go:211]: But why using len of push options?

**Review by a1012112796 — APPROVED** [review]
LGTM. by the way, suggest add this test function also.

```GO
func TestAgitPullPush(t *testing.T) {
	onGiteaRun(t, func(t *testing.T, u *url.URL) {
		baseAPITestContext := NewAPITestContext(t, "user2", "repo1", auth_model.AccessTokenScopeWriteRepository, auth_model.AccessTokenScopeWriteUser)

		u.Path = baseAPITestContext.GitPath()
		u.User = url.UserPassword("user2", userPassword)

		dstPath := t.TempDir()
		doGitClone(dstPath, u)(t)

		gitRepo, err := git.OpenRepository(context.Background(), dstPath)
		assert.NoError(t, err)
		defer gitRepo.Close()

		doGitCreateBranch(dstPath, "test-topic")

		// commit 1
		_, err = generateCommitWithNewData(littleSize, dstPath, "user2@example.com", "User Two", "branch-data-file-")
		assert.NoError(t, err)

		// push to create a agit pull request
		err = git.NewCommand(git.DefaultContext, "push", "origin", "HEAD:refs/for/master/test-topic").Run(&git.RunOpts{Dir: dstPath})
		assert.NoError(t, err)

		// check pull request exist
		unittest.AssertExistsAndLoadBean(t, &issues.PullRequest{BaseRepoID: 1, Flow: issues.PullRequestFlowAGit, HeadBranch: "user2/test-topic"})

		// commit 2
		_, err = generateCommitWithNewData(littleSize, dstPath, "user2@example.com", "User Two", "branch-data-file-2-")
		assert.NoError(t, err)

		// push 2
		err = git.NewCommand(git.DefaultContext, "push", "origin", "HEAD:refs/for/master/test-topic").Run(&git.RunOpts{Dir: dstPath})
		assert.NoError(t, err)

		// reset to first commit
		err = git.NewCommand(git.DefaultContext, "reset", "--hard", "HEAD~1").Run(&git.RunOpts{Dir: dstPath})
		assert.NoError(t, err)

		// test force push without confirm
		stderr := strings.Builder{}
		err = git.NewCommand(git.DefaultContext, "push", "origin", "HEAD:refs/for/master/test-topic").Run(&git.RunOpts{Dir: dstPath, Stderr: &stderr})
		assert.Error(t, err)
		assert.Contains(t, stderr.String(), "[remote rejected] HEAD -> refs/for/master/test-topic (request `force-push` push option)")

		// test force push with confirm
		err = git.NewCommand(git.DefaultContext, "push", "origin", "HEAD:refs/for/master/test-topic", "-o", "force-push").Run(&git.RunOpts{Dir: dstPath, Stderr: &stderr})
		assert.NoError(t, err)
	})
}

```

File: routers/private/hook_post_receive.go (hunk 1 above)
Comment (a1012112796) [reply routers/private/hook_post_receive.go:211]: > But why using len of push options?

I think it's better set a length when we know the max length of a slice.

**Comment by wxiaoguang** [comment]
> LGTM. by the way, suggest add this test function also.

Great, thank you for the awesome test.

File: routers/private/hook_post_receive.go (hunk 1 above)
Comment (wxiaoguang) [reply routers/private/hook_post_receive.go:211]: But there could be far more push options than it really needs here.

At most, here it only needs 2, but there could be 4 or 5 or even more push options. They are not related.

**Comment by lunny** [comment]
docs update https://gitea.com/gitea/docs/pulls/80

**Pull request merged**